### PR TITLE
add helm options of tolerations and affinity for istiod.

### LINKED ISSUE #26522: Standalone Istio Operator does not apply tolerations defined for istiod
**Bug description**
When installing Istio using the operator and IstioOperator custom resource applied, the operator does not honor tolerations applied to pilot (i.e., istiod). This issue is not observed when applying the same IstioOperator custom resource applied via `istioctl manifest apply` command; Tolerations are applied correctly when Installing Istio with CLI.

Example IstioOperator structure:
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    pilot:
      k8s:
        tolerations:
          - key: dedicated
            operator: Equal
            value: istio
```

[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Tolerations should be applied regardless of how IstioOperator resource is applied (`istioctl manifest apply` or Istio Operator) 

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
control plane version: 1.6.7
data plane version: 1.6.7 (6 proxies)
```

**How was Istio installed?**
Istio Operator:
```
istioctl operator init --operatorNamespace istio-system
```

**Environment where bug was observed (cloud vendor, OS, etc)**
N/A

### PR DESCRIPTION
**Pull request by morvencao**
resolve: https://github.com/istio/istio/issues/26522 and https://github.com/istio/istio/issues/29642

[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml [hunk 1]
```diff
@@ -889,6 +896,39 @@ spec:
       - name: config-volume
         configMap:
           name: istio
+      affinity:
+        nodeAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:901]: why do we need required *and* preferred?

Comment (morvencao) [reply manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:901]: Both the "hard" and "soft" requirements are put here to ensure istiod will be scheduled to node with listed arch, this also align with other components(istio-ingress/egress gateways), see: https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/templates/_affinity.tpl#L11-L43

Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:901]: Doesn't the hard requirement take precedence? I am not sure how the soft one can ever be used

Comment (morvencao) [reply manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:901]: We can specify different `weight` for each soft requirement, then the istiod pod will be scheduled to node with highest **weight** that matched the hard requirement.

**Review by howardjohn — APPROVED** [review]

**Pull request closed without merging**